File: ABC/abc024/abc024_b 2.py
#100点解法
max_t = 1100001
close = [0] * max_t #close[i] := 時刻iの時点でドアが何秒後に閉まる予定か？

N, T = map(int, input().split())
A = set(int(input()) for _ in range(N))

ans = 0
for t in range(1, max_t):
    if t in A:
        close[t] = T
    else:
        close[t] = max(0, close[t-1] - 1)
    
    ans += (close[t] > 0)

print(ans)

File: ABC/abc024/abc024_b 2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 100点解法
    int max_t = 1100001, N, T, a, ans = 0;
    set<int> A;
    vector<int> close(max_t, 0); // close[i] := 時刻iの時点でドアが何秒後に閉まる予定か？

    cin >> N >> T;
    for (int i = 0; i < N; i++) {
        cin >> a;
        A.insert(a);
    }

    for (int t = 1; t < max_t; t++) {
        if (A.find(t) != A.end()) close[t] = T;
        else close[t] = max(0, close[t-1] - 1);

        ans += (close[t] > 0);
    }
    cout << ans << endl;
}